app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt

```diff
@@ -525,6 +525,17 @@ class ThreadListFragment : TwoPaneFragment(), SwipeRefreshLayout.OnRefreshListen
 
             waitingBeforeNotifyAdapter = threadListViewModel.isRecoveringFinished
 
+            deletedItemsIndices = { indices ->
+                if (isMultiSelectOn) {
+                    val previousThreads = threadListAdapter.dataSet.filterIsInstance<Thread>()
+                    var isRemoved = false
+                    indices.forEach {
+                        isRemoved = mainViewModel.selectedThreads.remove(previousThreads[it])
+                    }
+                    if (isRemoved) publishSelectedItems()
```

Comment:
remove() only returns false if the object wasn't in the data list, but with our code here, if only one of the selected thread isn't in the selection, then we don't publish and all the other selected thread won't be deleted in the UI

Reply:
True. It's fixed.